sdk/provisioning/Azure.Provisioning/src/ProvisioningPlan.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Azure.Provisioning.Expressions;
+using Azure.Provisioning.Primitives;
+
+namespace Azure.Provisioning;
+
+// This is a draft to explore convenience APIs to make it easier to build tooling
+// around Azure infrastructure.  It's likely going to change a fair bit in the near future.
+
+/// <summary>
+/// Represents a composed collection of resources that can be compiled to bicep
+/// source, saved to disk, compiled to an ARM template, linted, validated, and
+/// deployed.
+/// </summary>
+public partial class ProvisioningPlan
```

Comment:
And using that `ProvisioningPlan` to do all sorts of fun stuff with your freshly composed resources.